src/transformers/modeling_utils.py

```diff
@@ -1364,7 +1364,7 @@ def _autoset_attn_implementation(
                 hard_check_only=False,
                 check_device_map=check_device_map,
             )
-        elif requested_attn_implementation in [None, "sdpa"]:
+        elif requested_attn_implementation in [None, "sdpa"] and not is_torch_tpu_available():
```

Comment:
This restricts using the model on CPU when `is_torch_tpu_available()` so a bit too restrictive maybe?

Reply:
Do you have a better idea how to fix this? I'm not familiar with this part of HF....

Reply:
cc @muellerzr

Reply:
Sorry for late reply. I was busy with something else. I guess this is exactly what my change does?